Added information about disabling backends

Yohello1 opened the pull request:
Hello, I just wanted to add this in since I often have to disable back ends for my builds. 
At some point I realised that someone who isn't as well versed in CMake as I am might have difficulty in figuring out how to disable the back ends quickly

illuhad commented:
Why do you often need to disable backends? AdaptiveCpp should automatically enable all available backends on the system, so most normal users should not have to bother with these variables, or am I missing something?

Just not sure whether they are important enough to have them in the main installation documentation :)

Yohello1 commented:
For some reason it tries to build for backends which dont exist on my system. So I have to manually disable them. Also sometimes I just don't need support for a certain backend on a system.

On another note, if I moved it to the advanced installation instructions, would that be more suiting?

illuhad commented:
> For some reason it tries to build for backends which dont exist on my system.

That shouldn't happen, can you provide some more information? Which backends does it try to build, and which ones do you actually have installed?

> On another note, if I moved it to the advanced installation instructions, would that be more suiting?

Yeah, that sounds good to me!

Yohello1 commented:
Tries to build with ROCm, I don't have it installed. 
I'll make the amendment to the documentation to re-arrange a few parts then, thank you!

illuhad commented:
It should only build ROCm if either it detects it, or perhaps there is something stuck in the cmake cache (maybe you used to have it installed earlier? Or maybe you explicitly enabled it at some point?)

See https://github.com/AdaptiveCpp/AdaptiveCpp/blob/e1b8b9b224ea9c16eb3622ccaad1b7fe30075a3e/CMakeLists.txt#L238
And this is how we detect ROCm: https://github.com/AdaptiveCpp/AdaptiveCpp/blob/e1b8b9b224ea9c16eb3622ccaad1b7fe30075a3e/CMakeLists.txt#L210

So if it enables ROCm without you having it installed, it would be a bug and not expected behavior.

Yohello1 committed:
Updated Install docs to easier to follow model

Yohello1 commented:
@illuhad Hi, I ended up just re-doing the entire change, hopfully this is better.
Sorry it's a bit large, I didn't really see a good way to split up the changes since it's kind of just a big re-org of how the installation docs.

Yohello1 committed:
Merge branch 'develop' into install_doc_modification

illuhad commented:
Oh wow...can you please provide a summary what exactly you changed and with which goal? It's a bit hard to review it otherwise.

Yohello1 commented:
Sorry about the really big PR, there wasn't really a way to split it up. I kinda just re-wrote all the installation docs because they were pretty hard to follow.
These changes kinda shift it towards: Read installing.md, then if you have a specific need, go read the install-[X].md about it, and just copy the flags from there when building.

Changes:

Installing.md:
- Added Quick Start section for people on Linux, who just want to get up and running
- Added section about OS support, what level different OS's recieve, and what backends they support (with links to relevant documentation)
- Pre-reqs listed, and cmake/python are now explicitly mentioned (I don't think they were before)
- If someone wants to use a specific backend, you now just look at the "Backend-Specific requirements" table, and look at the linked documentation as to how to install it & their general pre-reqs. Before it was just a not-well formated link to another document
- Building from source part has general overview, and a few basic flags which people may want/need

install-llvm:
- Easier to follow, and has a bit more info about installing/building it. 

install-metal.md:
- Re-formatted it, and removed info which was present in installing.md. Removing redundant info was a big theme in this, as it was sometimes confusing, and yeah.

advanced-builds.md:
- Moved stuff about 2-stage bootstrap, part about building LLVM with AdaptiveCPP as apart of it there.

install-cuda.md:
- Removed redundant info & formatted differently. 

install-ocl.md:
- Removed redundant info & formatted differently.  
- Spirv-LLVM-translator being automatically fetched is now a bit more noticeable
 
install-rocm.md:
- Removed redundant info & formatted differently.  

install-spirv.md:
- Removed redundant info & formatted differently.  

Sorry about the really big PR

Yohello1 commented:
@illuhad Hii, sorry to ping again, just wondering if you've had time to read over this PR yet?

Yohello1 committed:
Merge branch 'develop' into install_doc_modification
